File: Data_structure/829/queue.cpp
#include <iostream>
using namespace std;

const int N = 1e5 + 10;
int m, x, h, t, q[N];
char op[10];

int main() {
    scanf("%d", &m);
    while (m--) {
        scanf("%s", op);
        if (op[0] == 'p' && op[1] == 'u') {
            scanf("%d", &x);
            q[t++] = x;
        } else if (op[0] == 'p') {
            h++;
        } else if (op[0] == 'e') {
            if (h == t) printf("YES\n");
            else printf("NO\n");
        } else {
            printf("%d\n", q[h]);
        }
    }
    return 0;
}

File: Data_structure/829/queue.py
if __name__ == "__main__":
    n = int(input())
    q = [0] * (10 ** 5 + 10)
    h, t = 0, 0
    for _ in range(n):
        line = input().split()
        if line[0] == "push":
            q[t] = int(line[1])
            t += 1
        elif line[0] == "pop":
            h += 1
        elif line[0] == "empty":
            if h == t:
                print("YES")
            else:
                print("NO")
        else:
            print(q[h])
